packages/@aws-cdk/aws-lambda-python/README.md

```diff
@@ -36,6 +36,32 @@ new PythonFunction(this, 'MyFunction', {
 });
 ```
 
+Custom Docker images for bundling dependencies can be specified by specifying additional `buildImageOptions`. If using a custom Docker image, please ensure that the dependencies are stored at `/var/dependencies` within the Docker image for them to be bundled into the Lambda asset.
+
+A different bundling Docker image `Dockerfile.build` can be specified as:
+
+```ts
+new PythonFunction(this, 'MyFunction', {
+   ...
+   dockerBuildImageOptions: {
```

Comment:
@setu4993 I think that I would prefer to adopt the same approach that is used by both `NodejsFunction` and `GoFunction` and just allow the user to supply their own `DockerImage`. i.e.

```ts
new lambda.PythonFunction(this, 'my-handler', {
  bundling: {
    dockerImage: DockerImage.fromBuild('/path/to/Dockerfile'),
  },
});
```

Reply:
Thanks for the feedback, @corymhall! I should have some space to work on this and update it over the weekend.

Reply:
@corymhall : Thinking through this a bit more, I'm open to switching to that structure, but I think it'd require quite a bit of refactoring of the current dependency installation system. I'm open to doing that (possibly in a separate PR first) and then inheriting those changes to support bundling here. Thoughts?

Reply:
@setu4993 If you are willing to work on that refactoring then I think it is a good idea! Before making this library stable we would want to bring it in line with the other two anyway.

Reply:
@corymhall : What do you think about https://github.com/aws/aws-cdk/pull/18082? The allure of getting `lambda-Python` to stable (finally!) was enough motivation :).